File: code/c/control-flow/test-jump.c
/* Program: test-jump.c */
#include <stdio.h>

int main()
{
    int i = 0;
    char ch;
    
    while(i < 10000000)
    {
        i++;
        
        // Skip all even numbers
        if ( i % 2 == 0 ) continue;
        
        printf("At %d\n", i);
        
        printf("Quit? [y/N]: ");
        scanf(" %c", &ch);
        
        if (ch == 'y' || ch == 'Y')
        {
            printf("Quitting loop...");
            break;
        }
    }
    
    // Ending function
    return 0;
    
    // Code cannot be reached!
    printf("This will never be printed!\n");
}

File: code/pascal/control-flow/TestJump.pas
program TestJump;

procedure Main();
var
  i: Integer = 0;
  ch: Char;
begin
    while i < 10000000 do
    begin
        i := i + 1;
        
        if i mod 2 = 0 then continue; // Skip all even numbers
        
        WriteLn('At ', i);
        
        Write('Quit? [y/N]: ');
        ReadLn(ch);
        
        if (ch = 'y') or (ch = 'Y') then
        begin
            WriteLn('Quitting loop...');
            break;  // End the loop
        end;
    end;
    WriteLn('Bye...');
    exit; // Exit function
    
    // Code cannot be reached!
    WriteLn('This will never be printed!');
end;

begin
  Main();
end.